depth_pid:
  ros__parameters:
    kp: -1.0
    ki: 0.0
    kd: 2000.0
    kaw: 0.0
    max: 25.0
    min: -25.0
    max_rate: 50
    setpoint: 20.0
    input_margin: 2.0
    input_pressure: bluerov2/pressure
    input_wrench: joy_wrench_stmp
    output_wrench: PID/depth
    status: True
    publish_statistics: True

n_state_publisher:
  ros__parameters:
    robot_name: bluerov2

n_thruster_manager:
  ros__parameters:
    thruster_prefix: t200

    # Thruster Force Axis
    #
    # Axis of the thruster force. Used only for visualization purposes
    thruster_force_axis: 'x'

    thruster_wrench_publisher_name: n_thruster_wrench

    # Sub Input Wrench Topic Name
    #
    # Name of the topic where the input wrench is published
    #
    sub_input_wrench_topic_name: PID/depth

    # Output Type:
    #
    #
    # Parameters:
    #   PWM     - one public topic with list of pwm values for thrusters set up on 
    #   SEP_PWM - every Thruster creates topic with pwm value
    #   RPM     - one public topic with list of rpm values for thrusters
    #   SEP_RPM - every Thruster creates topic with rpm value
    #   FRC     - one public topic with list of force values for thrusters
    #   SEP_FRC - every Thruster creates topic with rpm value
    #   SEP_WRENCH
    #   STONEFISH - normalized values for stonefish
    #
    #  Order of thrusters is sorted alphabetically
    output_type: 'STONEFISH'

    # Thruster Voltage
    #
    # Voltage of the thrusters will have an impact on the thrust generated by the thrusters simulated in stonefish
    #
    # Possible values:
    #   10
    #   12
    #   14
    #   16
    #   18
    #   20
    thruster_voltage: 20

n_wrench_system_base:
  ros__parameters:
    controller: "ds4"

    wrench_topic_name: "joy_wrench"
    wrench_stamped_topic_name: "joy_wrench_stmp"

    send_stamped: True
    frame_id: base_link

    # Max Force
    #
    # Maximum force that can be applied by the controller
    # If given force is higher than calculated by thruster manager, it will be clipped
    max_force: 25.0 # kg f
    max_torque: 5.0 # Nm

    equalization_type: "cube"

    ds4_force_x: axis_left_y
    ds4_force_y: axis_left_x
    ds4_force_z: axis_right_y

    ds4_torque_x: axis_right_x
    ds4_torque_y: None
    ds4_torque_z: None

    inv_force_x: False
    inv_force_y: True
    inv_force_z: False

    inv_torque_x: True
    inv_torque_y: False
    inv_torque_z: True



